docs/adr/0001-tenants-will-not-inherit-their-parents-properties.md

```diff
@@ -0,0 +1,39 @@
+# Tenants will not inherit their parents properties
+
+## Status
+
+Proposed
+
+## Context
+
+Corvus.Tenancy supports a hierarchy of tenants. There are two things that this can be used for:
+
+Firstly, we can control on a per-tenant basis where the children of that tenant are stored. For example, with two sibling tenants A and B (i.e. tenants that are children of the same parent tenant), the children of A can be stored in a completely different location to the children of B. By default, this will be a separate container in the same storage account, but it could be a completely separate storage account.
```

Comment:
>where the children of that tenant are stored

I'm wondering if we might need to be more precise, because it's awfully easy to get confused between tenanted storage and tenant storage.

I tripped up on this sentence because I think of the "tenant" as being an entity that may be entirely outside of the system. Contoso is a tenant. Litware is a tenant. (Some tenants happen to be internal too; `Marain.Workflow` is a tenant, `Marain.Operations` is a tenant. But even here, the tenant is still an entity that is essentially external to `Corvus.Tenancy`.)

So I am uncomfortable saying that an `ITenant` instance _is_ the tenant. This may sounds like splitting hairs, but in this case, I think there is an important distinction to be made between the tenant and the object that represents the tenant. And the main reason for this is that the way we use this in Marain, we may actually have multiple different `ITenant` instances all corresponding to the same actual tenant, because of the on-behalf-of tenants we create when one service uses another service as part of its implementation.

So there isn't necessarily a one-to-one mapping between tenants of a multi-tenants service and `ITenant` instances.

On the other hand, that's a detail of how we use tenancy in Marain, so we don't want to bake that particular set of usage conventions into `Corvus.Tenancy`. So possibly we don't need to change it. But there is something about the phrase "where the children of that tenant are stored" that doesn't quite sit right with me, because tenant is invariably an overloaded term. What we actually store is a) properties, and b) a list of children.

Reply:
Have reworded slightly to try and make this distinction.